src/kudu/consensus/persistent_vars.h

```diff
@@ -0,0 +1,114 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+#pragma once
+
+#include <atomic>
+#include <cstdint>
+#include <deque>
+#include <string>
+
+#include "kudu/consensus/persistent_vars.pb.h"
+#include "kudu/consensus/quorum_util.h"
+#include "kudu/gutil/macros.h"
+#include "kudu/gutil/ref_counted.h"
+#include "kudu/gutil/threading/thread_collision_warner.h"
+
+namespace kudu {
+
+class FsManager;
+class Status;
+
+namespace consensus {
+
+class PersistentVarsManager; // IWYU pragma: keep
+class PersistentVarsTest;    // IWYU pragma: keep
+
+enum class PersistentVarsCreateMode {
```

Comment:
does this add any value, or is this a pattern copied from others?

Reply:
Borrowed from ConsensusMetadataManager

Reply:
I think no point carrying out extra functionality which we don't think we need.
Right now the code seems like a exact copy of consensus meta. I think FLUSH_ON_CREATE seems
like the right behavior.